integration-tests/src/test/burn.test.ts

```diff
@@ -0,0 +1,117 @@
+import { Blockchain, KeyringPair, creditcoinApi, BN } from 'creditcoin-js';
+import { forElapsedBlocks, testData } from 'creditcoin-js/lib/testUtils';
+import { CreditcoinApi } from 'creditcoin-js/lib/types';
+import { mnemonicGenerate } from '@polkadot/util-crypto';
+import { extractFee, describeIf } from '../utils';
+
+describeIf((global as any).CREDITCOIN_EXECUTE_SETUP_AUTHORITY, 'burn', () => {
+    let ccApi: CreditcoinApi;
+    let sudoSigner: KeyringPair;
+    let wallet: KeyringPair;
+
+    const testingData = testData(
+        (global as any).CREDITCOIN_ETHEREUM_CHAIN as Blockchain,
+        (global as any).CREDITCOIN_CREATE_WALLET,
+    );
+    const { keyring } = testingData;
+
+    beforeAll(async () => {
+        ccApi = await creditcoinApi((global as any).CREDITCOIN_API_URL);
+        sudoSigner = (global as any).CREDITCOIN_CREATE_SIGNER(keyring, 'lender');
+        wallet = (global as any).CREDITCOIN_CREATE_SIGNER(keyring, 'borrower');
+    });
+
+    afterAll(async () => await ccApi.api.disconnect());
+
+    it('burn_all works as expected', async (): Promise<void> => {
+        const {
+            api,
+            extrinsics: {},
+        } = ccApi;
+
+        await api.tx.sudo
+            .sudo(api.tx.balances.setBalance(wallet.address, new BN('1000000000000000000', 10), 0))
+            .signAndSend(sudoSigner, { nonce: -1 });
+        await forElapsedBlocks(api);
+
+        const starting = await api.derive.balances.all(wallet.address);
+        expect(starting.freeBalance.isZero()).toBe(false);
+
+        await api.tx.creditcoin.burnAll(sudoSigner.address).signAndSend(wallet);
+        await forElapsedBlocks(api);
+
+        const ending = await api.derive.balances.all(wallet.address);
+        expect(ending.freeBalance.isZero()).toBe(true);
+    }, 100_000);
+
+    it('burn works as expected', async (): Promise<void> => {
+        const burner = keyring.addFromMnemonic(mnemonicGenerate(12));
+
+        const {
+            api,
+            extrinsics: {},
+        } = ccApi;
+
+        await api.tx.sudo
+            .sudo(api.tx.balances.setBalance(burner.address, new BN('1000000000000000000', 10), 0))
+            .signAndSend(sudoSigner, { nonce: -1 });
+        await forElapsedBlocks(api);
+
+        const starting = await api.derive.balances.all(burner.address);
+        expect(starting.freeBalance.isZero()).toBe(false);
+
+        await api.tx.creditcoin.burn(100, sudoSigner.address).signAndSend(burner);
```

Comment:
The value `100` is that 100 CTC or is it 100 CREDO ? 

Maybe you can use `POINT_01_CTC` for readability as well.